### DISCUSSION TITLE
Proposal for Upgrading to Mainnet

### DISCUSSION THREAD

**Discussion by kostko** [oasisprotocol/community-forum#1] [Mainnet]
This document provides an overview of the proposed criteria and changes for Mainnet. This should be seen as a framework that the community can use to propose a specific time and process for Mainnet Launch.

# Current Status of the Oasis Network

The Oasis Network is currently in Mainnet Beta. In short, this is a near-complete decentralized network that includes a live ROSE token and operator incentives. ROSE token transfers on are disabled on Mainnet Beta.

# Criteria for Mainnet

In order to transition from Mainnet Beta to Mainnet, community members have collectively suggested the following criteria be met. This is a collection of community feedback.

* [x] Validators representing more than 2/3 of stake in the initial consensus committee successfully get online to launch Mainnet Beta.
* [x] Beta network runs successfully for at least 10 days.
* [x] No major security risks on the Beta Network have gone unremediated, and untested on the testnet for 10 days.
* [x] At least 50 validators run on the Network.
  * Throughout Mainnet Beta there have been between 75 and 77 active validators on the network.
* [x] There are NO Oasis Protocol Foundation or Oasis Labs nodes serving as validators.
* [x] At least one block explorer exists to track network stability, transactions, and validator activity.
  * There is much more. See [Block Explorers & Validator Leaderboards](https://docs.oasis.dev/general/community-resources/community-made-resources#block-explorers-validator-leaderboards) part of our docs.
* [x] At least one qualified custodian supports the native ROSE token.
  * Currently, Anchorage and Finoa support the ROSE token. See [Custody Providers](https://docs.oasis.dev/general/use-your-tokens/holding-tokens/custody-providers) part of our docs.
* [x] At least one web wallet or hardware wallet supports native ROSE token.
  * Currently, Bitpie mobile wallet and RockX Ledger-backed web wallet are available and support ROSE token transfers. Support for staking and delegation is in development. See [Mobile Wallets](https://docs.oasis.dev/general/use-your-tokens/mobile-wallets) and [Web Wallets](https://docs.oasis.dev/general/use-your-tokens/web-wallets) parts of our docs.

# Mechanics of Upgrading to Mainnet

Upgrading from Mainnet Beta to Mainnet will require a coordinated upgrade of the Network. All nodes will need to configure a new genesis file that they can generate or verify independently and reset/archive any state from Mainnet Beta. Once enough (representing 2/3+ of stake) nodes have taken this step, the network will start.

# Proposed Changes From Mainnet Beta to Mainnet

The Mainnet genesis file is intended to be as close as possible to the state of the Mainnet Beta network at the time of upgrade. That includes retaining validator token balances, retaining genesis file wallet allocations, and block height at time of the snapshot.

In addition, after receiving additional feedback from the community, the Oasis Protocol Foundation has proposed to increase the staking rewards model. In the new proposed model staking rewards will start at 20% (annualized) and range from 20% to 2% over the first 4 years of the network (see more in updated [Token Metrics and Distribution](https://docs.oasis.dev/oasis-network-primer/token-metrics-and-distribution) doc).

The following parts of the genesis file will be updated:

* `height` will be updated to be the block height at the time of the snapshot on Mainnet Beta.
* `genesis_time` will be updated with the time of the Mainnet launch.
* `chain_id` will be set to `oasis-1`.
* `halt_epoch` will be set to `9940` (approximately 1 year from Mainnet launch).
* `staking.params.disable_transfers` will be set to `false` (or omitted) to enable transfers.
* `staking.params.reward_schedule` will be updated to reflect the updated reward schedule as mentioned above.
* Transfer from the Community and Ecosystem Wallet (`staking.ledger.oasis1qrad7s7nqm4gvyzr8yt2rdk0ref489rn3vn400d6.general.balance`) of 450M ROSE to the Common Pool (`staking.common_pool`) to fund increased staking rewards.

Mainnet will use [**Oasis Core 20.12.2**](https://github.com/oasisprotocol/oasis-core/releases/tag/v20.12.2).

# Changelog

* 2020-11-13 - Changed the recommended Oasis Core version to 20.12.2.
* 2020-11-10 - Updated `chain_id` to `oasis-1` as per discussion.

**Comment by cfl0ws** [comment]
Chainflow generally supports this upgrade. We did have some questions and comments related to it - 

> At least one qualified custodian supports the native ROSE token.

Who is this?

>  At least one web wallet or hardware wallet supports native ROSE token.

Which wallets are these?

> chain_id will be set to mainnet-2020-11-xx-xxxxxxxxxx.

We'd suggest a simpler format, e.g. mainnet-01.

> halt_epoch will be set to 9940 (approximately 1 year from Mainnet launch).

This seems to imply that a new network will be launched within a year, is this the intention?

**Comment by wjdfx** [comment]
Bit Cat supports this upgrade.

BTW:
> Chainflow generally supports this upgrade. We did have some questions and comments related to it -
> 
> > At least one qualified custodian supports the native ROSE token.
> 
> Who is this? 

https://docs.oasis.dev/general/wallet-support/holding-rose-tokens

>
> > At least one web wallet or hardware wallet supports native ROSE token.
> 
> Which wallets are these?

https://docs.oasis.dev/general/use-your-tokens/web-wallets
https://docs.oasis.dev/general/use-your-tokens/mobile-wallets

> 
> > chain_id will be set to mainnet-2020-11-xx-xxxxxxxxxx.
> 
> We'd suggest a simpler format, e.g. mainnet-01.
> 
> > halt_epoch will be set to 9940 (approximately 1 year from Mainnet launch).
> 
> This seems to imply that a new network will be launched within a year, is this the intention?

**Comment by tjanez** [comment]
> 
> > At least one qualified custodian supports the native ROSE token.
> 
> Who is this?
> 
> > At least one web wallet or hardware wallet supports native ROSE token.
> 
> Which wallets are these?
>

As @wjdfx answered, this is in our docs, but we can copy it to issue's description so it's clearer which these are.

> > chain_id will be set to mainnet-2020-11-xx-xxxxxxxxxx.
> 
> We'd suggest a simpler format, e.g. mainnet-01.

Seems reasonable. What do others think?

> 
> > halt_epoch will be set to 9940 (approximately 1 year from Mainnet launch).
> 
> This seems to imply that a new network will be launched within a year, is this the intention?

We expect in the first year of the network that there will be many proposed upgrades and each would have a new halt epoch.

The goal of adding this 1 year limit is to ensure the community proposes at least 1 upgrade (but we expect more than that) in the first year.

**Comment by jcliff** [comment]
Scalar Capital supports this upgrade + the proposed criteria.

**Comment by pkrasam** [comment]
We support this upgrade and look forward to hearing more about the launch date, so as to plan for it.

**Comment by Bambarello** [comment]
Support the upgrade and the criteria.

Team, please try to have **non-ledger web-wallet** ready by the time of mainnet start as well.

**Comment by 1xCL** [comment]
> > > chain_id will be set to mainnet-2020-11-xx-xxxxxxxxxx.
> > 
> > 
> > We'd suggest a simpler format, e.g. mainnet-01.
> 
> Seems reasonable. What do others think?

I would suggest `mainnet-1a` in case we need to generate a new genesis file or `mainnet-1`.

**Comment by Afauvre** [comment]
@Bambarello please check out Bitpie. https://bitpie.com/

**Comment by kaustubhkapatral** [comment]
Witval is in support for this upgrade

**Comment by Bambarello** [comment]
> @Bambarello please check out Bitpie. https://bitpie.com/

Hey @Afauvre, yes, I checked, but this is not (yet) pure platform-independent web-wallet, it's IOS or Android based app, so not fully free of Apple/Google gegemony :)

**Comment by Afauvre** [comment]
> > @Bambarello please check out Bitpie. https://bitpie.com/
> 
> Hey @Afauvre, yes, I checked, but this is not (yet) pure platform-independent web-wallet, it's IOS or Android based app, so not fully free of Apple/Google gegemony :)

Oh I see what you're saying. There are a few other options in the works but i think they will come a little after Mainnet. Soon though and understand the feedback.

**Comment by franono** [comment]
> > > > chain_id will be set to mainnet-2020-11-xx-xxxxxxxxxx.
> > > 
> > > 
> > > We'd suggest a simpler format, e.g. mainnet-01.
> > 
> > 
> > Seems reasonable. What do others think?
> 
> I would suggest `mainnet-1a` in case we need to generate a new genesis file or `mainnet-1`.

I agree with changing the chain-id to something more readable such as mainnet-1a or oasis-mainnet-1a, simply to reduce confusion / silly mistakes in future upgrades or when troubleshooting issues.

**Comment by UbikCapital** [comment]
Looks good for us.

**Comment by joesixpack** [comment]
👍

**Comment by maryshann** [comment]
Blockdaemon supports the upgrade 👍

**Comment by oceanabcde** [comment]
Ocean Stake supports the upgrade. 👍👍👍

**Comment by spdd** [comment]
The upgrade and the criteria looks good for me. 🚀🚀🚀

**Comment by m4dbi7** [comment]
Stakefish support the upgrade, just the same opinion about chain-id to be something like "mainnet-1".

**Comment by liminal-state** [comment]
Chorus One supports this upgrade. We are working to get Ledger delegations and transfers working on Anthem until genesis time.

We agree with the notion of a succinct chain_id, and would, in light of a future, interoperable "Internet of Blockchains" suggest to simply use `oasis-1`. Same goes for multi-chain block explorers etc., Hubble e.g. is using the chain_id in the URL, so having mainnet-1 wouldn't be very distinct in that context.

**Comment by perfectstake** [comment]
Perfect Stake supports the upgrade.

**Comment by fmanea** [comment]
Ankr supports the upgrade

**Comment by keefertaylor** [comment]
Tessellated Geometry supports this upgrade proposal.

**Comment by fmanea** [comment]
I would propose another criteria:

At least one web wallet that supports account creation without a Ledger. This web wallet should support account recovery with mnemonic.

Even though it's not that secure, but it clearly would help for a more widespread adoption.

**Comment by jjangg96** [comment]
0base.vc supports the upgrade.

**Comment by Julia-Ju-Ju** [comment]
I support the upgrade, agree with @fmanea it's good to have web-wallet without a ledger at the mainnet start, but shall not delay that.

**Comment by lyoungblood** [comment]
Blockscale supports this upgrade. Go Oasis Protocol!

**Comment by Afauvre** [comment]
> I would propose another criteria:
> 
> At least one web wallet that supports account creation without a Ledger. This web wallet should support account recovery with mnemonic.
> 
> Even though it's not that secure, but it clearly would help for a more widespread adoption.

Thanks @fmanea does Bitpie suffice --> https://bitpie.com/

**Comment by fmanea** [comment]
@Afauvre,
Bitpie is a mobile wallet (and not a web wallet). 

On Bitpie you Setup your wallet with a mnemonic for all the assets in the Wallet ( BTC, ETH, TRX, ETC) and all the keys are derived somehow from the same 12-word Mnemonic, But i doubt that will be backwards competible outside the Mobile Wallet. You also don't have any mechanism of exporting the Oasis private key and use it somewhere else except the mobile app.

To give an example of what I was thinking, a good example would be how Avalanche did it on the mnemonic and private key manangemt side: https://wallet.avax.network/

Also, a good example of Staking within the wallet could be Near, where you can select to stake your tokens from a list of Validators - and all the process is done inside the wallet. (stake / unstake) - https://wallet.near.org

**Comment by Bambarello** [comment]
I fully support @fmanea in his suggestion, and @Afauvre kindly note, there is no (yet) fully independent oasis web-based wallet, which can be accessed from anywhere in browser with send/receive/stake functionality.

Bitpie you're referring to is not web-wallet, it's mobile wallet hosted by bitpie (so not fully independent).

**Comment by ank-everstake** [comment]
Everstake supports the upgrade.

**Comment by viktorbunin** [comment]
We approve as well with the nit that it is impossible to know if any major security risks have been undiscovered, as they have not yet been identified. May be worth updating that verbiage :)

**Comment by kostko** [comment]
> Chorus One supports this upgrade. We are working to get Ledger delegations and transfers working on Anthem until genesis time.
> 
> We agree with the notion of a succinct chain_id, and would, in light of a future, interoperable "Internet of Blockchains" suggest to simply use `oasis-1`. Same goes for multi-chain block explorers etc., Hubble e.g. is using the chain_id in the URL, so having mainnet-1 wouldn't be very distinct in that context.

Simplifying the chain ID field makes sense. Note that in our case the actual unique network identifier which is used in all the network protocols for domain separation is the [hash of the canonical genesis document](https://docs.oasis.dev/oasis-core/high-level-components/index/genesis) and not the chain ID directly. Although I agree that having a nice chain ID for display purposes is useful.

I've updated the initial post to include the new chain ID field.

**Comment by tjanez** [comment]
@fmanea, @Bambarello, thanks for voicing your support for the need for an easy to use web-wallet that also has the ability to derive a private key by itself (besides working with a hardware wallet like Ledger) so it is completely self-contained.

I want to add a little background why this hasn't been a higher priority for us.
We've been a bit hesitant to develop and endorse such a wallet because using it on a network-connected computer would greatly increase the risk of users' private keys being compromised. The attack surface is very broad, it could be a [0-day web browser exploit](https://blog.coinbase.com/responding-to-firefox-0-days-in-the-wild-d9c85a57f15b) or a [backdoor in a JS library](https://www.theregister.com/2019/06/07/komodo_npm_wallets/).

> 
> To give an example of what I was thinking, a good example would be how Avalanche did it on the mnemonic and private key manangemt side: https://wallet.avax.network/
> 
> Also, a good example of Staking within the wallet could be Near, where you can select to stake your tokens from a list of Validators - and all the process is done inside the wallet. (stake / unstake) - https://wallet.near.org

Thanks for these 2 examples, they look pretty slick.

Again, thanks for this feedback. We've noted it and will look into prioritizing this in the future.

**Comment by SimplyStaking** [comment]
Simply Staking votes in favour of the upgrade proposal on the November 18 at 4pm UTC.

**Comment by tjanez** [comment]
> Simply Staking votes in favour of the upgrade proposal on the November 18 at 4pm UTC.

Speaking for the Oasis Foundation, this sounds good. If there are no objects, we will proceed with this time for the upgrade to the Mainnet.

**Comment by gavinly** [comment]
LGTM. Figment excited to launch the upgrade on Nov 18 🚀

**Comment by kostko** [comment]
Mainnet was successfully launched on Nov 18th, well done and thank you everyone! :tada:
